src/pybox/local.py

```diff
@@ -218,25 +241,53 @@ def __init__(
         kernel_manager: MultiKernelManager | None = None,
         async_kernel_manager: AsyncMultiKernelManager | None = None,
         profile_dir: str | None = None,
+        log=None,
     ):
+        log = log if log else logger
         self.profile_dir = profile_dir
-
         if kernel_manager is None:
-            self.kernel_manager = MultiKernelManager()
+            self.kernel_manager = MultiKernelManager(log=log)
         else:
             self.kernel_manager = kernel_manager
         if async_kernel_manager is None:
-            self.async_kernel_manager = AsyncMultiKernelManager()
+            self.async_kernel_manager = AsyncMultiKernelManager(log=log)
         else:
             self.async_kernel_manager = async_kernel_manager
 
-    # TODO: I cannot use __del__ in async context
-    # maybe I shouldn't clean up the kernels in the __del__ method
-    def __del__(self):
+        # TODO: It works well in async scenarios but blocks in sync scenarios, so it should be disabled now.
+        # weakref.finalize(self, self.cleanup)
+
+    async def acleanup(self, now: bool = False):  # noqa: FBT002 FBT001
```

Comment:
You can force it to be a kwargs by adding '*, ' before `now` to overcome the ruff warnings.